File: src/main/java/com/intellij/tapestry/core/util/chain/Chain.java
package com.intellij.tapestry.core.util.chain;

/**
 * An ordered sequence of {@link Command}s that is itself a {@link Command}.
 */
public interface Chain extends Command {

    /**
     * Appends a command to this chain.
     *
     * @param command the command to add.
     */
    void addCommand(Command command);
}

File: src/main/kotlin/com/github/rar91279/plugin/tapestry/core/util/chain/Chain.kt
package com.github.rar91279.plugin.tapestry.core.util.chain

/**
 * An ordered sequence of [Command]s that is itself a [Command].
 */
interface Chain : Command {

    /**
     * Appends a command to this chain.
     */
    fun addCommand(command: Command)
}
